python/lsst/sphgeom/_healpixPixelization.py

```diff
@@ -0,0 +1,174 @@
+        # Shift back to the original resolution and uniquify
+        pixels = np.unique(np.right_shift(neighbors, self._bit_shift))
+
+        return RangeSet(pixels)
+
+    def interior(self, region: Region, maxRanges: int = 0):
+        if maxRanges > 0:
+            # If this is important, the rangeset can be consolidated.
+            raise NotImplementedError("HealpixPixelization: maxRanges not implemented")
+        pixels = self._interior_pixels_from_region(self._nside, region)
+
+        # Check that the corners of the pixels are entirely enclosed in
+        # the region
+
+        # Returns array [npixels, 3, ncorners], where ncorners is 4, and
+        # the center index points to x, y, z.
+        corners = hp.boundaries(self._nside, pixels, step=1, nest=True)
+
+        corners_int = region.contains(corners[:, 0, :].ravel(),
+                                      corners[:, 1, :].ravel(),
+                                      corners[:, 2, :].ravel()).reshape((len(pixels), 4))
+        interior = (np.sum(corners_int, axis=1) == 4)
+        pixels = pixels[interior]
+
+        return RangeSet(pixels)
+
+    def _interior_pixels_from_region(self, nside: int, region: Region):
+        """Get interior pixels from a region.
+
+        Parameters
+        ----------
+        nside : `int`
+            Healpix nside to retrieve interior pixels.
+        region : `lsst.sphgeom.Region`
+            Sphgeom region to find interior pixels.
+
+        Returns
+        -------
+        pixels : `np.ndarray`
+            Array of pixels at resolution nside, nest ordering.
+        """
+        _circle = None
+        _poly = None
+        if isinstance(region, (Box, Ellipse)):
+            _circle = region.getBoundingCircle()
+        elif isinstance(region, Circle):
+            _circle = region
+        elif isinstance(region, ConvexPolygon):
+            _poly = region
+        else:
+            raise ValueError("Invalid region.")
+
+        # Find all pixels at an arbirarily higher resolution
+        if _circle is not None:
+            center = _circle.getCenter()
+            vec = np.array([center.x(), center.y(), center.z()]).T
+            pixels = hp.query_disc(nside,
+                                   vec,
+                                   _circle.getOpeningAngle().asRadians(),
+                                   inclusive=False,
+                                   nest=True)
+        else:
+            vertices = np.array([[v.x(), v.y(), v.z()] for v in _poly.getVertices()])
+            pixels = hp.query_polygon(nside,
+                                      vertices,
+                                      inclusive=False,
+                                      nest=True)
+
+        return pixels
+
+    def __eq__(self, other):
+        if isinstance(other, HealpixPixelization):
+            return (self._level == other._level)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
```

Comment:
I think Python now takes care of `__ne__` for you if you implement `__eq__` (it didn't always, but I think this has been true for a long time).